indico/modules/rb_new/controllers/user/landing.py

```diff
@@ -16,10 +16,53 @@
 
 from __future__ import unicode_literals
 
+import dateutil
+from flask import request
+
+from indico.core.db import db
+from indico.modules.rb import rb_settings
 from indico.modules.rb.controllers import RHRoomBookingBase
+from indico.modules.rb.models.reservations import RepeatFrequency
+from indico.modules.rb.models.rooms import Room
 from indico.modules.rb_new.views.base import WPRoomBookingBase
+from indico.util.string import natural_sort_key
 
 
 class RHRoomBookingLanding(RHRoomBookingBase):
     def _process(self):
         return WPRoomBookingBase.display('room_booking.html')
+
+
+class RHRoomBookingSearch(RHRoomBookingBase):
+    def _process(self):
+        from indico.modules.rb.schemas import rooms_schema
+
+        min_capacity = int(request.args['capacity']) if 'capacity' in request.args else None
+        room_name = request.args.get('title')
+        start_dt = dateutil.parser.parse(request.args['start_dt'])
+        end_dt = dateutil.parser.parse(request.args['end_dt'])
```

Comment:
Can't `start_dt` and `end_dt` be `None` somehow?

Reply:
I would rather make it mandatory, `start_dt`, at least to me, should be a must. For `end_dt` we could always assume that it is the same as `start_dt` if `None`